test/unit/event/src/EventTest.cpp

```diff
@@ -227,44 +236,260 @@ TEMPLATE_LIST_TEST_CASE("waitForEventThatAlreadyFinishedShouldBeSkipped", "[even
             alpaka::Event<Queue> e1(f1.m_dev);
 
             // 1. kernel k1 is enqueued into queue q1
-            // q1 = [k1]
             alpaka::enqueue(q1, k1);
+            // q1 = [k1]
+
             // 2. kernel k2 is enqueued into queue q2
-            // q2 = [k2]
             alpaka::enqueue(q2, k2);
+            // q2 = [k2]
 
             // 3. event e1 is enqueued into queue q1
-            // q1 = [k1, e1]
             alpaka::enqueue(q1, e1);
+            // q1 = [k1, e1]
 
             // 4. q2 waits for e1
-            // q2 = [k2, ->e1]
             alpaka::wait(q2, e1);
+            // q2 = [k2, ->e1]
 
             // 5. kernel k1 finishes
-            // q1 = [e1]
             k1.trigger();
+            // q1 = [e1]
 
             // 6. e1 is finished
-            // q1 = []
             alpaka::wait(e1);
+
+            // q1 = []
+            // q2 = [k2, ->e1]
+            REQUIRE(!alpaka::isComplete(k2));
             REQUIRE(alpaka::isComplete(e1));
 
             // 7. e1 is re-enqueued again but this time into q2
-            // q2 = [k2, ->e1, e1]
             alpaka::enqueue(q2, e1);
 
+            // q2 = [k2, ->e1, e1]
+            REQUIRE(!alpaka::isComplete(k2));
+            REQUIRE(!alpaka::isComplete(e1));
+
             // 8. kernel k2 finishes
-            // q2 = [->e1, e1]
             k2.trigger();
+            // q2 = []
+            REQUIRE(alpaka::isComplete(e1));
```

Comment:
Shouldn't we wait for `e1` here before asking if it is ready?

Reply:
`trigger()` will add `200ms` wait, after that time the state of the event should be set correctly.
If we would add a wait here and someone is refactoring something which let the test fail we will produce a never terminating (will terminate after the timeout) test. This will be extremely hard to debug compared to a clear answer from the system that the event is in the wrong state.

see also https://github.com/alpaka-group/alpaka/pull/2058#discussion_r1284372985
The test is checking that all is in an explicit state.